### PR TITLE
reduce the network traffic for single-namespace tests

### PR DESCRIPTION
**Pull request by jmuk**
This changes the behavior of CRD store to watch only the specified
namespace if namespace restriction is set. This is a feature
to be used by tests only, and this reduces the network traffic
significantly, when multiple tests are running in a single cluster.

This does not affect production use cases.

Note that this limits the mixer CRD store to watch only one namespace;
previously it can watch a list of namespaces. However, I haven't seen
any use cases of watching to a list of namespaces. CRD store would
watch the all namespaces (released ones, cluster-wide), or watch
a namespace for e2e tests. This behavior simply satisfies the current
use cases.

### CODE REVIEW

File: mixer/pkg/config/crd/store.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func waitForSynced(ctx context.Context, informers map[string]cache.SharedInforme
 // Store offers store.StoreBackend interface through kubernetes custom resource definitions.
 type Store struct {
 	conf         *rest.Config
-	ns           map[string]bool
+	ns           string
```
Comment (andraxylia) [inline mixer/pkg/config/crd/store.go]: cc @ayj If I remember correctly this was added so the CRD controller can watch a list of namespaces. Is this behavior preserved by the changes in the PR?

**Comment by andraxylia** [comment]
can you validate it still works for cluster-wide by installing istio in istio-system and bookinfo in default namespace? (Unfortunately there is no automated test).

File: mixer/pkg/config/crd/store.go (hunk 1 above)
Comment (jmuk) [reply mixer/pkg/config/crd/store.go]: Yes, this changes the behavior, from list of namespaces to a single namespace.
Justification: so far I don't see any use cases to watch multiple namespaces. e2e tests specifies a single namespace, and others (like cluster-wide tests and release version) watch all namespaces. The new code supports both, and no one needs to list up multiple namespaces so far. I think the existing scheme was overly generalized.

Added this note to the PR description.

**Comment by jmuk** [comment]
@andraxylia `make e2e_mixer` with `--cluster_wide` succeeds on my environment.

**Comment by jmuk** [comment]
PTAL @mandarjog @andraxylia

**Comment by jmuk** [comment]
@mandarjog please review this

**Pull request closed without merging**